### ISSUE TITLE
ImmerComponentStore: defaultState (via constructor) is not immutable

### ISSUE THREAD

**Issue by SebastianPost1996** [timdeschryver/ngrx-immer#26]
When initializing a store and defining the initial state in the constructor, that initial store is not produced by immer and thus not immutable until a state change has been performed.

```ts
interface TestState {
  test: string;
}

@Injectable({ providedIn: 'root' })
export class TestStore extends ImmerComponentStore<TestState> {
  constructor() {
    super({ test: 'foo' });
    this.get().test = 'bar'; // should throw an error but does not
  }
}
```

Replacing the defaultState in the `super` call by a produced value correctly causes an error to be thrown.

```
super(produce({ test: 'foo' }, x => x));
```

**Comment by timdeschryver** [comment]
Thanks for raising this issue, do you want to create a PR for this @SebastianPost1996 ?

**Comment by SebastianPost1996** [comment]
Added PR: https://github.com/timdeschryver/ngrx-immer/pull/27

**Comment by timdeschryver** [comment]
Closes #27
